### PR TITLE
Fix Flysystem integration and add tests

### PR DESCRIPTION
**Pull request by Lctrs**

### CODE REVIEW

**Comment by Lctrs** [comment]
Still WIP, I didn't fixed streams' handling.

**Comment by Lctrs** [comment]
I have honestly no idea how to fix streams' handling since Flysystem doesn't support append operations or creating a writable stream.

**Comment by bytehead** [comment]
We have a StreamWrapper feature in our [FlySystemBundle](https://github.com/1up-lab/OneupFlysystemBundle/blob/master/Resources/doc/filesystem_stream_wrapper.md). Did you check this?

PS: I don't know what is broken anyway - can you explain?

**Comment by Lctrs** [comment]
2 things : 

1) We call `fopen` without giving full path to the file, so it always fail because the filesystem's root is likely not in the same dir as the class.
2) We call `League\Flysystem\Filesystem::putStream()` which doesn't append but create or overwrite an existing file with the given content (unless you created a special Filesystem for the chunks and gave it `FILE_APPEND` write flags in the constructor but we can't check for it).

**Comment by Lctrs** [comment]
But I think this can effectively be fixed with the StreamWrapper feature. I'll do some tests.

**Comment by Lctrs** [comment]
@bytehead Should be good now.

File: Tests/Uploader/Storage/FlysystemOrphanageStorageTest.php [hunk 1]
```diff
@@ -38,12 +39,14 @@ public function setUp()
             $this->markTestSkipped('Temporary directories do not match');
```
Comment (bytehead) [inline Tests/Uploader/Storage/FlysystemOrphanageStorageTest.php]: Test is still skipped...

Comment (bytehead) [reply Tests/Uploader/Storage/FlysystemOrphanageStorageTest.php]: Let's see if tests still pass 😄

**Comment by Lctrs** [comment]
@bytehead Any comments ? :)

**Comment by bytehead** [comment]
I get some errors on my local environment. Need to debug first. Strangely the tests on travis pass...

**Comment by Lctrs** [comment]
Tests are maybe incomplete.

I tested my work locally with Fine Uploader with direct upload and chunked upload with differents Flysystem's Filesystem for chunks and final medias and everything worked like a charm.

**Comment by bytehead** [comment]
Alright, thanks for testing! I'll merge.

**Pull request merged**